### PR TITLE
DP-379 Feature/add ros pkg pipeline

### PR DESCRIPTION
**Pull request by duartecoelhomovai**
Deploy our new ros package pipeline template. To check my validations go to 
https://github.com/MOV-AI/movai_odom_utils/actions

- [x] Make sure you are opening from a **topic/feature/bugfix branch**
- [x] Ensure that the PR title represents the desired changes
- [x] Ensure that the PR description detail the desired changes
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

[^note]:
    Put an `x` into the [ ] to show you have filled the information.
    The template comes from https://github.com/MOV-AI/.github/blob/master/.github/pull_request_template.md
    You can override it by creating .github/pull_request_template.md  in your own repository

### CODE REVIEW

File: .github/workflows/ros-pkg-workflow.yml [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+
+        search_result=$(cat $0 | grep "$1")
+
+        if [ -z "$search_result" ]:
+        then
+            echo "\e[31m$0"
+            touch $2
+        fi 
+
+        ' {} $expected_url $VALIDATION_FILE \;
```
Comment (AlexFernandes-MOVAI) [inline .github/workflows/ros-pkg-workflow.yml:69]: please simplify

**Review by AlexFernandes-MOVAI — CHANGES_REQUESTED** [review]

File: .github/workflows/ros-pkg-workflow.yml [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+          -Dsonar.qualitygate.timeout=300
+
+    - name: Link to SonarQube Dashboard
+      shell: bash
+      run: |
+        echo "Please check report here: https://sonarqube.aws.cloud.mov.ai/dashboard?id=${{ github.event.repository.name }}"
+      id: sonar_links
+
+    - name: Install Mobros
+      run: python3 -m pip install mobros==1.0.1.8
```
Comment (AlexFernandes-MOVAI) [inline .github/workflows/ros-pkg-workflow.yml:100]: can we insert that inside the docker image instead ?

Comment (duartecoelhomovai) [reply .github/workflows/ros-pkg-workflow.yml:100]: In my opinion, would be easier to simply bump here as this is the one that has the main behavior should mainly be covered by mobros. With this we can simply bump the template with a new behavior without the need to rebuild the docker image. 
mobros install time = 5 secs,
docker image build time = 5 mins +  (plus promotion and so on)

File: .github/workflows/ros-pkg-workflow.yml (hunk 1 above)
Comment (duartecoelhomovai) [reply .github/workflows/ros-pkg-workflow.yml:69]: Also fix the problem of submodules. We can only validate package.xmls that are ours. Either check maintainer mail or package name containing movai OR create a new extra attribute in the export section.

**Review by AlexFernandes-MOVAI — COMMENTED** [review]
Lot of code duplication, can we factorize it in .github ?

File: .github/workflows/ros-pkg-workflow.yml (hunk 2 above)
Comment (AlexFernandes-MOVAI) [reply .github/workflows/ros-pkg-workflow.yml]: ok

**Pull request merged**